### ISSUE TITLE
@starting-style doesn't nest correctly

### ISSUE THREAD

**Issue by mariusGundersen** [less/less.js#4267]
**To reproduce:**

[Playground link](http://lesscss.org/less-preview/#eyJjb2RlIjoiI2xpYigpIHtcbiAgICAuY29sb3JzKCkge1xuICAgICAgQHByaW1hcnk6IGJsdWU7XG4gICAgICBAc2Vjb25kYXJ5OiBncmVlbjtcbiAgICB9XG4gICAgLnJ1bGVzKEBzaXplKSB7XG4gICAgICBib3JkZXI6IEBzaXplIHNvbGlkIHdoaXRlO1xuICAgIH1cbiAgfVxuICBcbiAgLmJveCB3aGVuICgjbGliLmNvbG9yc1tAcHJpbWFyeV0gPSBibHVlKSB7XG4gICAgd2lkdGg6IDEwMHB4O1xuICAgIGhlaWdodDogKCR3aWR0aCAvIDIpO1xuICB9XG4gIFxuICAuYmFyOmV4dGVuZCguYm94KSB7XG4gICAgQG1lZGlhIChtaW4td2lkdGg6IDYwMHB4KSB7XG4gICAgICB3aWR0aDogMjAwcHg7XG4gICAgICAjbGliLnJ1bGVzKDFweCk7XG4gICAgfVxuICB9IiwiYWN0aXZlVmVyc2lvbiI6IjQuMi4wIiwibWF0aCI6InBhcmVucy1kaXZpc2lvbiIsInN0cmljdFVuaXRzIjpmYWxzZX0=)

```less
[popover]:popover-open {
  opacity: 1;
  transform: scaleX(1);

  @starting-style {
    opacity: 0;
    transform: scaleX(0);
  }
}
```

**Current behavior:**

This produces the following output, which is clearly incorrect:

```css
[popover]:popover-open {
  opacity: 1;
  transform: scaleX(1);
}
@starting-style {
  opacity: 0;
  transform: scaleX(0);
}
```

**Expected behavior:**

It should produce the following css:

```css
[popover]:popover-open {
  opacity: 1;
  transform: scaleX(1);

  @starting-style {
    opacity: 0;
    transform: scaleX(0);
  }
}
```

**Environment information:**

- `less` version: 4.2.0

This is based on the sample code from mdn: https://developer.mozilla.org/en-US/docs/Web/CSS/@starting-style

**Comment by rejhgadellaa** [comment]
With more and more @-rules being added (e.g. `@starting-style` and `@position-try`) and browsers supporting nested css (and thus @-rules), I think the default behavior of Less should indeed be changed. Should we move this into a separate ticket?

**Comment by rejhgadellaa** [comment]
Additional information:

`@starting-style` is now supported on [Chromium and WebKit](https://web.dev/blog/web-platform-05-2024#the_starting-style_rule). Firefox has it implemented and [plans to enable it](https://bugzilla.mozilla.org/show_bug.cgi?id=1900458) soon.

**Comment by shrpne** [comment]
`@starting-style` is supported in Firefox now from 129 version

**Comment by mariusGundersen** [comment]
> Experimental CSS, by definition, is generally not supported by Less.

Starting style is not experimental anymore

**Comment by ogdakke** [comment]
Yup, this should be supported now. Any timeline on this?

**Comment by puckowski** [comment]
I have working code for ```@starting-style``` support on top of Less 4.2.1 (4.2.1 is yet to be published to npm) on my machine. I'll get the support into my fork https://github.com/puckowski/less.js. The fork also has support for the ```@scope``` at-rule. ```master``` branch will be clear when ```@starting-style``` support is ready.

If things go well I'll prepare a PR to this repo, but no guarantee of code merge or that a release will be published.

I maintain my fork since I use it for some of my projects, so PRs are welcome there.

**Comment by puckowski** [comment]
Less.js 4.2.1 fork release with support for CSS Container Queries, Media Queries Level 4, starting-style at-rule, and scope at-rule support with all tests passing: https://github.com/puckowski/less.js/releases/tag/4.2.1-container-mq4-tr2-scope-1

I'll likely get around to submitting a PR to this repo just for ```@starting-style``` next weekend.

If you use the fork and run into an issue, feel free to open an issue on the fork.

**Comment by puckowski** [comment]
@matthew-dean 

Thank you for publishing Less.js ```4.2.1```. 

I do not have access otherwise I would help, but release ```4.2.1``` doesn't look published here on GitHub Releases, just FYI. I know some users have asked about that in the past.

**Comment by puckowski** [comment]
PR #4289 should resolve this issue. I'll support any requested revisions until it gets merged.

**Comment by jpattishall-ebay** [comment]
Hi @puckowski, out of curiosity, will this also address `@container` in `@media` not properly nesting?

Currently I see in 4.2.1 [Playground](https://lesscss.org/less-preview/#eyJjb2RlIjoiQG1lZGlhIG9ubHkgc2NyZWVuIGFuZCAobWluLXdpZHRoOiA3NjhweCkge1xuICBAY29udGFpbmVyIChtaW4td2lkdGg6IDUwMHB4KSB7XG4gICAgLnByaW1hcnktY29udGVudCB7XG4gICAgICBmb250LXNpemU6IDFyZW07XG4gICAgfVxuICB9XG59IiwiYWN0aXZlVmVyc2lvbiI6IjQuMi4xIiwibWF0aCI6InBhcmVucy1kaXZpc2lvbiIsInN0cmljdFVuaXRzIjpmYWxzZX0=) I see this:
```
@media only screen and (min-width: 768px) {
  @container (min-width: 500px) {
    .primary-content {
      font-size: 1rem;
    }
  }
}
```

results in:
```
@container only screen and (min-width: 768px) and (min-width: 500px) {
  .primary-content {
    font-size: 1rem;
  }
}
```

But I see [4.1.3](https://lesscss.org/less-preview/#eyJjb2RlIjoiQG1lZGlhIG9ubHkgc2NyZWVuIGFuZCAobWluLXdpZHRoOiA3NjhweCkge1xuICBAY29udGFpbmVyIChtaW4td2lkdGg6IDUwMHB4KSB7XG4gICAgLnByaW1hcnktY29udGVudCB7XG4gICAgICBmb250LXNpemU6IDFyZW07XG4gICAgfVxuICB9XG59IiwiYWN0aXZlVmVyc2lvbiI6IjQuMS4zIiwibWF0aCI6InBhcmVucy1kaXZpc2lvbiIsInN0cmljdFVuaXRzIjpmYWxzZX0=) produced the expected output:

```
@media only screen and (min-width: 768px) {
  @container (min-width: 500px) {
    .primary-content {
      font-size: 1rem;
    }
  }
}
```
Thanks!

**Comment by puckowski** [comment]
@jpattishall-ebay PR #4289 does not resolve/address the issue you are describing. Thank you for your comment and for raising that separate issue. I was not previously aware of that issue.

I've taken a look at your issue on Less.js 4.2.1 and I believe I have a fix, but I need some time for further testing and revision. I'll try to prepare a separate PR by Friday 12/20/24.

I'll likely get my fix into my fork and cut a release before I file the PR, so I will notify you if you'd like to use that while waiting on the official Less.js release.

**Comment by puckowski** [comment]
I put together a new PR for this issue with a different approach. See https://github.com/less/less.js/pull/4333

**Comment by mariusGundersen** [comment]
I'm leaving this issue open but I'm unsubscribing as we have decided to migrate away from less because of this issue. Now that css supports nesting and less doesn't support modern css there weren't any good reasons to stay with less.

**Comment by puckowski** [comment]
This issue should be resolved via release ```4.3.0``` available on npm: https://www.npmjs.com/package/less/v/4.3.0

Thank you for the report.
